### PR TITLE
elasticbeanstalk - add eb environment resource

### PR DESCRIPTION
**Pull request by jniesen**
~This commit adds the `elasticbeanstalk-environment` resource as well as
an `environment-uptime` filter for this resource.~

This commit adds the `elasticbeanstalk-environment` resource and verifies that the `regex` operator and `age` value type can be used to filter out unfit environments.

### CODE REVIEW

File: c7n/resources/elasticbeanstalk.py [hunk 1]
```diff
@@ -32,3 +35,48 @@ class resource_type(object):
...
+            'DateCreated',
+            'DateUpdated',
+            )
+        filter_name = 'EnvironmentNames'
+        filter_type = 'list'
+
+    filter_registry = eb_env_filters
+
+@eb_env_filters.register('environment-uptime')
+class EnvironmentUptimeFilter(AgeFilter):
```
Comment (kapilt) [inline c7n/resources/elasticbeanstalk.py]: we actually folded the date comparison functionality directly into the value filter with value_type: age  and just point `key` at any date attribute. so we really don't need a separate filter for it to write a policy like the example here.

Comment (jniesen) [reply c7n/resources/elasticbeanstalk.py]: Oh sweet! I'll remove that then. Thanks for the quick response.

**Comment by jniesen** [comment]
@kapilt could you take a look at those failing builds? The freezegun library was added in 3ba6c32, but it doesn't seem to be in the version of `requirements-dev.txt` that the CI servers are building. Not sure how to go about troubleshooting this on my end.

**Comment by kapilt** [comment]
not sure whats the issue with freeze_gun... 

we already have a way of doing this via mocking, see the test_offhours (and others) for usage around freezing time for date comparisons with mock_datetime_now

**Comment by kapilt** [comment]
i've made the nesc changes to remove the freezegun dependency over here, https://github.com/kapilt/cloud-custodian/tree/jniesen-eb_inactive_terminatea

**Comment by jniesen** [comment]
Thank you @kapilt. I'll pull your change in and update the PR.

**Pull request merged**